dataplane/handlers/interface.go

```diff
@@ -224,26 +224,40 @@ func (ni *Interface) handleLinkUpdate(lu *netlink.LinkUpdate) {
 	iface := ni.getOrCreateInterface(modelName)
 	iface.GetOrCreateEthernet().MacAddress = ygot.String(lu.Attrs().HardwareAddr.String())
 	iface.Ifindex = ygot.Uint32(uint32(lu.Attrs().Index))
-	iface.Enabled = ygot.Bool(lu.Attrs().Flags == net.FlagUp)
+	iface.Enabled = ygot.Bool(lu.Attrs().Flags&net.FlagUp != 0)
+	iface.AdminStatus = oc.Interface_AdminStatus_DOWN
+	if *iface.Enabled {
+		iface.AdminStatus = oc.Interface_AdminStatus_UP
+	}
+	// TODO: handle other states.
 	var operStatus oc.E_Interface_OperStatus
-	switch lu.Attrs().OperState { // TODO: handle other states.
+	switch lu.Attrs().OperState {
 	case netlink.OperDown:
 		operStatus = oc.Interface_OperStatus_DOWN
-	case netlink.OperUp:
+	case netlink.OperUp, netlink.OperUnknown: // TAP interfaces are in unknown state until there is socket bound to them.
```

Comment:
where does the socket get attached? Is there just two ends to the TAP interface?

Reply:
by some other process,